File: dart/test/angle_test.dart
/* This is free and unencumbered software released into the public domain. */

import 'dart:math' as math;

import 'package:test/test.dart';

import 'package:dogma/dogma.dart' show Angle;

void main() {
  group('Angle', () {
    test('Angle.fromDegrees() constructs the angle from degrees', () {
      expect(Angle.fromDegrees(0).degrees, equals(0));
      expect(Angle.fromDegrees(90).degrees, equals(90));
      expect(Angle.fromDegrees(180).degrees, equals(180));
      expect(Angle.fromDegrees(360).degrees, equals(360));
    });

    test('Angle.fromRadians() constructs the angle from radians', () {
      expect(Angle.fromRadians(0).radians, equals(0));
      expect(Angle.fromRadians(0.5 * math.pi).radians, equals(0.5 * math.pi));
      expect(Angle.fromRadians(math.pi).radians, equals(math.pi));
      expect(Angle.fromRadians(2 * math.pi).radians, equals(2 * math.pi));
    });

    test('Angle.fromTurns() constructs the angle from turns', () {
      expect(Angle.fromTurns(0).turns, equals(0));
      expect(Angle.fromTurns(0.25).turns, equals(0.25));
      expect(Angle.fromTurns(0.5).turns, equals(0.5));
      expect(Angle.fromTurns(1).turns, equals(1));
    });
  });
}

File: go/dogma/angle_test.go
// This is free and unencumbered software released into the public domain.

package dogma

import "math"
import "testing"
import "github.com/stretchr/testify/assert"

func TestAngleFromDegrees(t *testing.T) {
	assert.Equal(t, 0.0, AngleFromDegrees(0).Degrees())
	assert.Equal(t, 90.0, AngleFromDegrees(90).Degrees())
	assert.Equal(t, 180.0, AngleFromDegrees(180).Degrees())
	assert.Equal(t, 360.0, AngleFromDegrees(360).Degrees())
}

func TestAngleFromRadians(t *testing.T) {
	assert.Equal(t, 0.0, AngleFromRadians(0).Radians())
	assert.Equal(t, 0.5*math.Pi, AngleFromRadians(0.5*math.Pi).Radians())
	assert.Equal(t, 1*math.Pi, AngleFromRadians(1*math.Pi).Radians())
	assert.Equal(t, 2*math.Pi, AngleFromRadians(2*math.Pi).Radians())
}

func TestAngleFromTurns(t *testing.T) {
	assert.Equal(t, 0.0, AngleFromTurns(0).Turns())
	assert.Equal(t, 0.25, AngleFromTurns(0.25).Turns())
	assert.Equal(t, 0.5, AngleFromTurns(0.5).Turns())
	assert.Equal(t, 1.0, AngleFromTurns(1).Turns())
}
